parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -69,6 +72,20 @@
   public static final byte[] MAGIC = "PAR1".getBytes(Charset.forName("ASCII"));
   public static final int CURRENT_VERSION = 1;
 
+  // need to supply a buffer size when setting block size. this is the default
+  // for hadoop 1 to present. copying it avoids loading DFSConfigKeys.
+  private static final int DFS_BUFFER_SIZE_DEFAULT = 4096;
+
+  // visible for testing
+  static final Set<String> BLOCK_FS_SCHEMES = new HashSet<String>();
+  static {
+    BLOCK_FS_SCHEMES.add("hdfs");
```

Comment:
Looks like [HFTP is a read-only FS|https://hadoop.apache.org/docs/r2.4.1/hadoop-project-dist/hadoop-hdfs/Hftp.html] so we shouldn't need to add that. But I think we should add viewfs as you suggested and webhdfs. Anything backed by HDFS should probably have it but we can add more over time.